include/soci/soci-unicode.h

```diff
@@ -0,0 +1,569 @@
+#ifndef SOCI_UNICODE_H_INCLUDED
+#define SOCI_UNICODE_H_INCLUDED
+
+#include "soci/error.h"
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <wchar.h>
+
+// Define SOCI_WCHAR_T_IS_WIDE if wchar_t is wider than 16 bits (e.g., on Unix/Linux)
+#if WCHAR_MAX > 0xFFFFu
+#define SOCI_WCHAR_T_IS_WIDE
```

Comment:
Hopefully it's not a difficult change to make, as it's just a global search-and-replace, but could we please rename this one to `SOCI_WCHAR_T_IS_UTF32`? Reading "wide char is wide" is quite confusing IMO.